src/sentry/tasks/backfill_seer_grouping_records.py

```diff
@@ -554,6 +551,114 @@ def delete_seer_grouping_records(
+    for group_ids_chunk in chunked(high_event_count_group_ids, SNUBA_QUERY_RATELIMIT):
+        snuba_requests = []
+        for group_id in group_ids_chunk:
+            group = Group.objects.get(id=group_id)
+            query = Query(
+                match=events_entity,
+                select=[
+                    Column("group_id"),
+                    Column("event_id"),
+                ],
+                where=[
+                    Condition(Column("project_id"), Op.EQ, project_id),
+                    Condition(Column("group_id"), Op.EQ, group_id),
+                    Condition(
+                        Column("timestamp", entity=events_entity),
+                        Op.GTE,
+                        group.last_seen - timedelta(minutes=5),
+                    ),
+                    Condition(
+                        Column("timestamp", entity=events_entity),
+                        Op.LT,
+                        group.last_seen + timedelta(minutes=5),
+                    ),
+                ],
+                limit=Limit(1),
+            )
+
+            request = Request(
+                dataset=Dataset.Events.value,
+                app_id=Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value,
+                query=query,
+                tenant_ids={
+                    "referrer": Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value,
+                    "cross_org_query": 1,
+                },
+            )
+            snuba_requests.append(request)
+
+        snuba_results = bulk_snuba_queries(
+            snuba_requests, referrer=Referrer.GROUPING_RECORDS_BACKFILL_REFERRER.value
+        )
+
+        if snuba_results and snuba_results[0].get("data"):
+            group_event_rows += (snuba_result["data"][0] for snuba_result in snuba_results)
+
+    return group_event_rows
+
+
+def bulk_query_events(
+    low_event_count_group_ids: list[int],
+    low_event_count_groups_last_seen: list[datetime],
+    project_id: int,
+) -> list[EventGroupSnubaResult]:
+    """
+    Query events for groups in bulk, using `low_event_count_groups_last_seen` to limit the
+    timestamp of the query.
+    """
+    events_entity = Entity("events", alias="events")
+    query = Query(
+        match=events_entity,
+        select=[
+            Column("group_id"),
+            Function("max", [Column("event_id")], "event_id"),
+        ],
+        groupby=[Column("group_id")],
+        where=[
+            Condition(Column("project_id"), Op.EQ, project_id),
+            Condition(Column("group_id"), Op.IN, low_event_count_group_ids),
+            Condition(
+                Column("timestamp", entity=events_entity),
+                Op.GTE,
+                min(low_event_count_groups_last_seen) - timedelta(minutes=5),
+            ),
+            Condition(
+                Column("timestamp", entity=events_entity),
+                Op.LT,
+                max(low_event_count_groups_last_seen) + timedelta(minutes=5),
```

Comment:
the regular bulk snuba event query, but modify the timestamp condition to only look between the lowest and highest `times_seen` of the batch (+- 5 mins) instead of querying all 90 days